### PR TITLE
Implement asymmetric neighbor detection

### LINKED ISSUE #17: Implement asymmetric neighbour detection
Implement asymmetric neighbour detection using reflection, requires #16

### PR DESCRIPTION
**Pull request by reisub-de**
Fixes #17 
Also more comments and docstrings

### CODE REVIEW

File: dmpr/config.py [hunk 1]
```diff
@@ -11,12 +15,12 @@
...
         {
-            "name": "wlan0", "addr-v4": "10.0.0.1",
-            "link-characteristics": {"bandwidth": 100000, "loss": 0}
+            "name": "wlan0", "addr-v4": "10.0.0.1", "asymm-detection": False,
+            "link-attributes": {"bandwidth": 100000, "loss": 0}
         },
         {
-            "name": "tetra0", "addr-v4": "10.0.0.1",
-            "link-characteristics": {"bandwidth": 10000, "loss": 0}
+            "name": "tetra0", "addr-v4": "10.0.0.1", "asymm-detection": True,
```
Comment (hgn) [inline dmpr/config.py:22]: `asym-detection` ? Or even asymmetric-detection?

File: dmpr/dmpr.py [hunk 2]
```diff
@@ -250,7 +174,7 @@ def msg_rx(self, interface_name: str, msg: dict):
 
         self.trace('rx.msg', msg)
 
-        if interface_name not in self._conf['interfaces']:
+        if interface_name not in self.interfaces:
```
Comment (hgn) [inline dmpr/dmpr.py:177]: Not sure, I separated all user configured values, without modification into self._conf. Just to make it clear to the audience that these values are raw configured values and strictly separated from generated values.

NACK, for this for now!

It is important that dmp-core implementation is *really* simple and understandable. Motto: try to keep the python skills to a bare minimum, it must be read like real pseudo code. That is really important here. And separating config values from other values makes the software peace more understandable.

File: dmpr/dmpr.py [hunk 3]
```diff
@@ -435,17 +370,17 @@ def _parse_msg_db(self, policy: AbstractPolicy) -> tuple:
...
         """
-        interface = self._conf['interfaces'][interface_name]
+        interface = self.interfaces[interface_name]
         path = Path(path=neighbor,
                     attributes=LinkAttributes(),
                     next_hop=neighbor,
                     next_hop_interface=interface_name)
 
-        path.append(self._conf['id'], interface_name,
+        path.append(self.id, interface_name,
```
Comment (hgn) [inline dmpr/dmpr.py:383]: Id is probably fine, an exception to my comment before!

Let me extend there previous comment: if the value is generated from a config value, then they belong somewhere else _config values are unmodified raw config values. The id can be generated (uuid, if no default is given).

**Review by hgn — COMMENTED** [review]
The code changes looks good to me. Can you take a look at the inline comment ... Please drop a line if you suggest to merge it ...

File: dmpr/dmpr.py (hunk 2 above)
Comment (reisub-de) [reply dmpr/dmpr.py:177]: fair enough, will revert this

**Comment by hgn** [comment]
Merged, thank you!

**Pull request merged**